Fix the ECMAScript edition

Alex-CS opened the pull request:
The link goes to the ECMAScript 2015 spec, which is actually the 6th edition.

timgraham commented:
I think it's likely that the link used to be to the 5th edition but changed when the 6th edition was released in June 2015. Keeping the text that says "5th edition of" but removing it from the link might make sense.

Alex-CS commented:
@timgraham, you're right, it looks like the page at that url is just always the newest specification, my bad. I've changed the text back to "5th edition" and changed the link to the section of ECMAScript 5.1 specification that talks about the Array constructor. I chose that link because it takes you directly to the relevant section whereas the 5th edition spec is only available in PDF, but I can also just use the PDF link for 5.0 if you'd rather.
